src/driver/mongodb.rs

```diff
@@ -22,7 +22,7 @@ impl Driver for MongoDriver {
   fn connect(uri: &str) -> Result<Self, Error> {
     let config = try!(connstring::parse(uri));
 
-    if let Some(db_name) = config.clone().database {
+    if let Some(db_name) = config.database.clone() {
```

Comment:
I'm pretty sure, look at the lines below it. We move `config` and then expect `db_name` to still be in scope.